src/main/java/com/atlassian/jira/cloud/jenkins/deploymentinfo/service/JiraDeploymentInfoSenderImpl.java

```diff
@@ -88,7 +93,19 @@ public JiraSendInfoResponse sendDeploymentInfo(final JiraDeploymentInfoRequest r
             return JiraDeploymentInfoResponse.failureEnvironmentInvalid(jiraSite, errorMessages);
         }
 
-        final Set<String> issueKeys = issueKeyExtractor.extractIssueKeys(deployment);
+        Set<String> jenkinsIssues = issueKeyExtractor.extractIssueKeys(deployment);
```

Comment:
@jonathanpbailey as I mentioned in my previous comment, this change makes perfect sense and we would be very interested in an actual usage of this change. Have you used it in your deployment pipeline? Would be able to describe your use case, i.e. how you obtain and store the information about the changeSet? Do you maintain this information between deployments to different environments, etc?

Reply:
Yes, we are currently using this in a CICD pipeline.  Our pipeline use case is as follows (trunk based):

1. merge/push master
2. auto deploy to staging
3. QA verifies
4. run Jenkins parameterized build to deploy.

Since the jira-software-cloud plugin only picks up changeset information on pushes and merges, the deployment information was never being gathered in step 4 and send to the Jira host.  So, even when a production deployment was made the associated Jira stories never registered the production deployment.

To facilitate this scenario I added an argument to the plugin to accept additional changeset information.  Then, in my pipeline (Jenkinsfile), I have it ascertain and pass all change sets between the last deployed release up to what is being deployed.